js/main.js

```diff
@@ -184,56 +188,81 @@ function saveVerse(event) {
+  entryObject.title = $verseTitle.textContent; // create a title property in entryObject; assign the textContent of $verseTitle to the title property
+  entryObject.id = data.EntryId; // create a id property in entryObject; assign the current entryid of the data model id property
 
-  data.view = 'journal-page';
-  viewSwapper();
+  var newEntry = renderEntry(entryObject); // render that entryObject so that it shows on the journal page
+  $ul.prepend(newEntry); // prepend the rendered object onto to the ul element
+
+  data.EntryId++; // increment the EntryId property by one so the next verse will be assigned this Id
+  data.entries.push(entryObject); // push that object into the entries property array of the data model
+
+  data.view = 'journal-page'; // set the data view to 'journal-page' so it only shows the journal entries
+  viewSwapper(); // run the viewSwapper function
+}
+
+document.addEventListener('DOMContentLoaded', loadedDOMContent);
+
+function loadedDOMContent(event) {
+  var $ul = document.querySelector('#saved-entries');
+  for (var i = 0; i < data.entries.length; i++) {
+    var renderedEntry = renderEntry(data.entries[i]);
+    $ul.append(renderedEntry);
+  }
 }
 
+// listens for clicks on the new/edit button while calling an anonymous function.
 document.addEventListener('click', function (event) {
-  data.view = 'edit-page';
-  var $ul = document.querySelector('#journal-entries');
-  var $textArea = document.querySelector('textarea');
-  $ul.className = 'padding-bottom';
-  if (event.target.textContent === 'Edit') {
-    viewSwapper();
-
-    if ($ul.hasChildNodes() === false) { // if this is a new entry...
-      var $li = document.createElement('li'); // create a dom tree with the verse you got
-      var $p = document.createElement('p');
-      $p.id = 'journal-verse';
-
-      $ul.appendChild($li);
-      $li.appendChild($p);
-      $p.textContent = data.entries[0].verse; // get the verse in the data entry and set it as the shown verse
-    } else { // if this is not a new entry or you're editing...
-      var currentDiv = event.target.closest('li').firstElementChild; // grab the closest li element and its first child which should be a div
-      var currentVerse = currentDiv.firstElementChild.textContent; // grab the first child of that div which should be a p element and get its text content
-      var $currentVerse = document.getElementById('journal-verse'); // grab the p elemennt that is used to show the verse
-      $currentVerse.textContent = currentVerse; // set the p element that shows the verse to the current verse you decided to journal on
-
-      if (data.entries[getEntryIndex()].entry === undefined) {
-        $textArea.value = '';
-      } else {
-        $textArea.value = data.entries[getEntryIndex()].entry;
-      }
+  if (event.target.textContent === 'Edit' || event.target.textContent === 'Write Journal Entry') { // if the user clicks the button and the text content says 'Edit'...
```

Comment:
Rather than basing this off of text content (what if a colleague wanted to change the words in the buttons again? What if that person didn't know the JavaScript depended on it?), I would add a class or something to state whether this verse has a journal entry yet or not.

Reply:
https://user-images.githubusercontent.com/97268025/157993646-5a13b4be-3d54-414c-89ce-e706321f58f9.mov

Reply:
https://user-images.githubusercontent.com/97268025/157997702-9f879cb2-bc4b-49f2-8f37-28b96eb933e1.mov